site/app/models/Gradeable.php

```diff
@@ -1339,4 +1339,181 @@ public function calculateLateStatus($checkTeam = true, $user_id = null) {
         }
     }
 
+    public function getGradingSectionsForUser(User $user) {
+        if ($this->getPeerGrading() && $user->getGroup() === 4) {
+            $users = $this->core->getQueries()->getPeerAssignment($this->getId(), $user->getId());
+            //TODO: Peer grading team assignments
+            return new GradingSection($this->core, $this, "Peer", [$user], $users, []);
+        } else {
+            $users = [];
+            $teams = [];
+
+            if ($this->isGradeByRegistration()) {
+                $section_names = $user->getGradingRegistrationSections();
+
+                if ($this->isTeamAssignment()) {
+                    foreach ($section_names as $section) {
+                        $teams[$section] = [];
+                    }
+                    $all_teams = $this->core->getQueries()->getTeamsByGradeableAndRegistrationSections($this->getId(), $section_names);
+                    foreach ($all_teams as $team) {
+                        /** @var Team $team */
+                        $teams[$team->getRegistrationSection()][] = $team;
+                    }
+                } else {
+                    foreach ($section_names as $section) {
+                        $users[$section] = [];
+                    }
+                    $all_users = $this->core->getQueries()->getUsersByRegistrationSections($section_names);
+                    foreach ($all_users as $user) {
+                        /** @var User $user */
+                        $users[$user->getRegistrationSection()][] = $user;
+                    }
+                }
+                $graders = $this->core->getQueries()->getGradersForRegistrationSections($section_names);
+            } else {
+                $section_names = $this->core->getQueries()->getRotatingSectionsForGradeableAndUser($this->getId(), $user->getId());
+
+                if ($this->isTeamAssignment()) {
+                    foreach ($section_names as $section) {
+                        $teams[$section] = [];
+                    }
+                    $all_teams = $this->core->getQueries()->getTeamsByGradeableAndRotatingSections($this->getId(), $section_names);
+                    foreach ($all_teams as $team) {
+                        /** @var Team $team */
+                        $teams[$team->getRotatingSection()][] = $team;
```

Comment:
Could avoid this loop if you juts made the return value of `getTeamsByGradeableAndRotatingSections` return the array indexed like this?

Reply:
The point of those queries was to be a Team counterpart to the User queries, which don't index by section. See 10 lines below this. I could likely convert both but I don't want to make them inconsistent for the sake of moving a for-loop into a different file.

Reply:
fair enough